### PR TITLE
Move field and scalar tests to their respective test modules (for #1106)

### PR DESCRIPTION
**Pull request by TeranIvy**
Moves tests for fields and scalars from `dynamo0p3.py` to their respective test modules in `src/psyclone/tests/domain/lfric/` (for #1106).

### CODE REVIEW

File: src/psyclone/tests/domain/lfric/lfric_field_test.py [hunk 1]
```diff
@@ -91,6 +96,22 @@
 '''
 
 
+def test_ad_fld_type_1st_arg():
```
Comment (arporter) [inline src/psyclone/tests/domain/lfric/lfric_field_test.py:99]: I think this is a step in the right direction but we're still ending up with a test file that is very big. I think we need to separate out those tests that are purely meta-data related and those that test the code generation.

File: src/psyclone/tests/domain/lfric/lfric_field_test.py [hunk 2]
```diff
@@ -233,7 +543,658 @@ def test_field_invoke_uniq_declns_valid_intrinsic():
...
+    field = kernel.arguments.args[3]
+    assert field.space == 'w2trace'
+    assert not field.discontinuous
+    # Test w2htrace
+    field = kernel.arguments.args[7]
+    assert field.space == 'w2htrace'
+    assert not field.discontinuous
+
+
+# Tests for PSy-layer code generation for field arguments
```
Comment (arporter) [inline src/psyclone/tests/domain/lfric/lfric_field_test.py:690]: So I think this bit ought to have its own test file...

File: src/psyclone/tests/domain/lfric/lfric_scalar_test.py [hunk 3]
```diff
@@ -105,25 +455,474 @@ def test_int_real_scalar_invalid():
             "This is invalid." in str(err.value))
 
 
-def test_scalar_invoke_uniq_declns_valid_intrinsic():
-    ''' Tests that all valid intrinsic types for user-defined scalar arguments
-    ('real' and 'integer') are accepted by Invoke.unique_declarations().
+# Tests for PSy-layer code generation for scalar arguments
```
Comment (arporter) [inline src/psyclone/tests/domain/lfric/lfric_scalar_test.py:458]: Again, I think we should keep the meta-data tests separate from the code-generation tests.

**Review by arporter — CHANGES_REQUESTED** [review]
Thanks for doing this tidying.
Modified files are pylint clean.
Total number of tests is conserved.
I think that, in order to reduce file size, it would be good to further split the tests according to whether they are metadata or code-generation related.

File: src/psyclone/tests/domain/lfric/lfric_field_test.py (hunk 1 above)
Comment (TeranIvy) [reply src/psyclone/tests/domain/lfric/lfric_field_test.py]: Split the field tests into `lfric_field_mdata_test.py` and `lfric_field_codegen_test.py`.

File: src/psyclone/tests/domain/lfric/lfric_field_test.py (hunk 2 above)
Comment (TeranIvy) [reply src/psyclone/tests/domain/lfric/lfric_field_test.py:62]: Done.

File: src/psyclone/tests/domain/lfric/lfric_scalar_test.py (hunk 3 above)
Comment (TeranIvy) [reply src/psyclone/tests/domain/lfric/lfric_scalar_test.py]: Split the scalar tests into `lfric_scalar_mdata_test.py` and `lfric_scalar_codegen_test.py`.

**Comment by TeranIvy** [comment]
GitHub now works so this PR is back with @arporter.

File: doc/user_guide/dynamo0p3.rst [hunk 4]
```diff
@@ -2034,28 +2034,6 @@ logic determined by their :ref:`access modes <dynamo0.3-kernel-valid-access>`.
   are updated (albeit in a different way due to different access to DoFs, see
   :ref:`dynamo0.3-api-meta-args` for more details).
 
-
```
Comment (arporter) [inline doc/user_guide/dynamo0p3.rst:2037]: This should not have been deleted?

File: doc/user_guide/examples.rst [hunk 5]
```diff
@@ -1,7 +1,7 @@
 .. -----------------------------------------------------------------------------
 .. BSD 3-Clause License
 ..
-.. Copyright (c) 2018-2021, Science and Technology Facilities Council.
+.. Copyright (c) 2018-2020, Science and Technology Facilities Council.
```
Comment (arporter) [inline doc/user_guide/examples.rst:4]: This file should not have been modified?

**Review by arporter — CHANGES_REQUESTED** [review]
I'm not going to do a full review as it appears that in bringing this branch up to master something has gone wrong with the examples and a lot of Joerg's recent changes have been reverted 
(e.g. examples/lfric/eg17 has been moved back to examples/lfric/full_example). Please could you investigate?

**Comment by TeranIvy** [comment]
@arporter, I think this is due to backing off and remerging the master to the branch so everything got mixed up. Shall I create a fresh branch/PR and port the changes there?

**Comment by arporter** [comment]
Hi @TeranIvy, it's up to you so whichever is easiest :-)

**Comment by TeranIvy** [comment]
@arporter, I tried another merge from master and it still doesn't fix the problem. I'll close this PR and create another branch.

**Pull request closed without merging**